CMakeLists.txt

```diff
@@ -211,12 +178,66 @@ if(FATAL_WARNINGS)
     #    warning C4245: '=': conversion from 'int' to 'uint64_t', signed/unsigned mismatch
     #    warning C4305: '=': truncation from 'double' to 'float'
         add_compile_options(/wd4100 /wd4505 /wd4702 /wd4389 /wd4245 /wd4305 /wd4018)
-    else()
-        add_compile_options(-Werror)
-    endif()
 endif()
 
+# NOTE: jsoncpp is located in SUBPROJECTS_DIR.
+set (SUBPROJECTS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external)
+
+find_path(JSONCPP_INCLUDE_DIR json/json.h HINTS "${SUBPROJECTS_DIR}/jsoncpp/dist"
+                                                   "${SUBPROJECTS_DIR}/JsonCpp/dist"
+                                                   "${SUBPROJECTS_DIR}/JsonCPP/dist"
+                                                   "${SUBPROJECTS_DIR}/JSONCPP/dist"
+                                                   "${CMAKE_SOURCE_DIR}/../jsoncpp/dist"
+                                             DOC "Path to jsoncpp/dist/json/json.h")
+
+find_path(JSONCPP_SOURCE_DIR jsoncpp.cpp HINTS "${SUBPROJECTS_DIR}/jsoncpp/dist"
+                                                   "${SUBPROJECTS_DIR}/JsonCpp/dist"
+                                                   "${SUBPROJECTS_DIR}/JsonCPP/dist"
+                                                   "${SUBPROJECTS_DIR}/JSONCPP/dist"
+                                                   "${CMAKE_SOURCE_DIR}/../jsoncpp/dist"
+                                             DOC "Path to jsoncpp/dist/jsoncpp.cpp")
+
+find_library(JSONCPP_LIB NAMES jsoncpp HINTS ${JSONCPP_SEARCH_PATH} )
+
+if (WIN32)
+    add_library(jsoncpp         STATIC IMPORTED)
+    find_library(JSONCPP_DLIB NAMES jsoncpp
+                 HINTS ${JSONCPP_DEBUG_SEARCH_PATH} )
+    set_target_properties(jsoncpp PROPERTIES
+                         IMPORTED_LOCATION       "${JSONCPP_LIB}"
+                         IMPORTED_LOCATION_DEBUG "${JSONCPP_DLIB}")
+endif()
+
+find_package(valijson REQUIRED CONFIG)
+
+# Always need C++2011 throughout the project.
+set(CMAKE_CXX_STANDARD 17)
+
+# FetchContent_MakeAvailable(jsoncpp) will set these values (which
+# affect the whole build).  Make sure they're set even if we
+# don't regenerate jsoncpp.
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib" CACHE PATH "Archive output dir.")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib" CACHE PATH "Library output dir.")
+set(CMAKE_PDB_OUTPUT_DIRECTORY     "${CMAKE_BINARY_DIR}/bin" CACHE PATH "PDB (MSVC debug symbol)output dir.")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin" CACHE PATH "Executable/dll output dir.")
```

Comment:
This is one of the benefits of using find_package over FetchContent. You don't have to do this anymore since jsoncpp will be in it's own build.